dd-sdk-android/src/main/kotlin/com/datadog/android/error/internal/DatadogExceptionHandler.kt

```diff
@@ -38,6 +40,11 @@ internal class DatadogExceptionHandler(
             RumErrorSource.SOURCE,
             e
         )
+        // make sure the Log ScheduledWriter finished
+        (writer as? ScheduledWriter)?.executorService?.awaitTermination(
```

Comment:
am I right that we rely on this to call executor shutdown: https://github.com/DataDog/dd-sdk-android/blob/5b609edbdcf75fd94b3653efdf5e9e494269a517/dd-sdk-android/src/main/kotlin/com/datadog/android/Datadog.kt#L100? but then mentioned hook will be called only after `previousHandler?.uncaughtException(t, e)` is called below. It means that this `awaitTermination` call will always throw `InterruptedException`, because of the timeout, since no `shutdown` command is triggered yet. Or I miss something?

Reply:
Ahh you are right, I was actually hopping that this to return if the Thread is idled but it seems it waits for that executor to shutdown. It will not throw an `InterruptedException` unless that thread is interrupted while waiting. In this case what is actually doing is waiting for 2 seconds and then unblocking the thread (more like a `Thread.sleep(2000)`) and is not exactly what I wanted. I need to find something else.